src/main/java/spireMapOverhaul/patches/interfacePatches/RewardModifierPatches.java

```diff
@@ -0,0 +1,127 @@
+                for (ArrayList<AbstractCard> cards : additionalCardRewards) {
+                    RewardItem rewardItem = new RewardItem();
+                    rewardItem.cards = cards;
+                    __instance.rewards.add(rewardItem);
+                }
+            }
+        }
+
+        private static class Locator extends SpireInsertLocator {
+            @Override
+            public int[] Locate(CtBehavior ctMethodToPatch) throws Exception {
+                Matcher matcher = new Matcher.NewExprMatcher(RewardItem.class);
+                int[] lines = LineFinder.findAllInOrder(ctMethodToPatch, matcher);
+                // We only want the first two uses, which are respectively the branch for the Vintage modifier (only
+                // elites and bosses drop cards) and the normal branch. The third use is for Prayer Wheel.
+                // The reason this matters (and why this isn't just a postfix patch) is because want to avoid adding
+                // additional card rewards to fights that normally don't drop rewards, and we don't want to reproduce
+                // the logic that CombatRewardScreen.setupItemRewards has for that.
+                return new int[] { lines[0], lines[1] };
+            }
+        }
+    }
+
+    @SpirePatch2(clz = AbstractRoom.class, method = "update")
+    public static class ModifyRewardsPatch {
+        @SpireInsertPatch(locator = Locator.class)
+        public static void ModifyRewards(AbstractRoom __instance) {
+            AbstractZone zone = Wiz.getCurZone();
+            if (zone instanceof IRewardModifyingZone) {
+                // We don't want to affect uses of CombatRewardScreen that are in non-battle events such as Sensory Stone
+                if (AbstractDungeon.getCurrMapNode() != null && AbstractDungeon.getCurrMapNode().room != null && AbstractDungeon.getCurrMapNode().room.isBattleOver) {
+                    IRewardModifyingZone z = ((IRewardModifyingZone)zone);
+                    for (RewardItem rewardItem : AbstractDungeon.combatRewardScreen.rewards) {
+                        if (rewardItem.type == RewardItem.RewardType.CARD && rewardItem.cards != null && !rewardItem.cards.isEmpty()) {
+                            z.modifyRewardCards(rewardItem.cards);
+                        }
+                        z.modifyReward(rewardItem);
+                    }
+                    AbstractDungeon.combatRewardScreen.rewards.addAll(z.getAdditionalRewards());
+                    z.modifyRewards(AbstractDungeon.combatRewardScreen.rewards);
+                }
+            }
+        }
+
+        private static class Locator extends SpireInsertLocator {
+            @Override
+            public int[] Locate(CtBehavior ctMethodToPatch) throws Exception {
+                // We ensure that this happens after the save file is made, outside of any checks for loading saves
```

Comment:
This is how we avoid save/reload inconsistency.

It's still possible to create save/reload inconsistency by using other RNG's in the card methods, because the weird thing about card rewards is that when you win a fight, the card rewards are created before the save file... but when you save and load, the card rewards are created again, based on the save file. 

The base game is able to do this without causing issues because it has special code to roll back the card RNG; see use of `AbstractDungeon.cardRng.counter` in `AbstractRoom.update` if you're curious.

The typical way things go wrong is when mods patch into the card creation logic and add other RNG calls. Effectively, that makes saving and loading advance RNG counters a second time. Depending on exactly what a mod is doing, there are other ways to get this right, but also other ways to get this wrong. It's a tricky area of code.